### DISCUSSION TITLE
Should KalendarEvent be an interface?

### DISCUSSION THREAD

**Discussion by morganesoula** [hi-manshu/Kalendar#65] [Q&A] [answered]
Hey there,

I see in your library that your main composable Kalendar can take a list of KalendarEvent to mark specific days. What does it mean? If I give it a list of events, will it be put on given day according to the field "date" (sorry I am not an English native speaker) or do you have another logic behind specific days?

What if I want to add some fields? Like a place or a time? Do you think it would be possible to discuss the fact that it could be an interface (then I would be able to extend it) rather than a data class?

**Comment by hi-manshu** [comment, marked as answer]
Hey hey!
Thank you for reaching out. To answer your question,
1. KalendarEvent has the events that will be displayed if passed for any specific date, like an event or any custom thing,
2. KalendarEvent being an interface can be a good idea or we can add a custom field for the calendar to be added by devs. 


Let me have a look into the API design
